core/node/vm_runner/src/impls/playground.rs

```diff
@@ -120,6 +138,23 @@ impl VmPlayground {
         ))
     }
 
+    async fn dump_vm_state(object_store: &dyn ObjectStore, dump: &VmDump) -> anyhow::Result<()> {
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("bogus clock");
+        let timestamp = timestamp.as_millis();
+        let batch_number = dump.l1_batch_number();
+        let dump_filename = format!("shadow_vm_dump_batch{batch_number:08}_{timestamp}.json");
```

Comment:
Not sure whether a timestamp suffix is required. Of course, its purpose is to retain dumps for unrelated errors in the same batch; but as a downside, if the same error occurs, it'll still be dumped to a multitude of files. I guess if a single dump / error handler is implemented (see the other line comment), it could be possible to use a tag obtained by hashing the error message (?).